File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/06_Nested_Loops_Exercise/04_Train_The_Trainers.js
function trainTheTrainers(input) {  // ["3","Arrays","4.53","5.23","5.00","Lists","5.83","6.00","5.42","Finish"]
    let judgesCount = Number(input[0]);
    let totalGradeSum = 0;
    let totalPresentationsCount = 0;
    // comment
    let i = 1;
    let currentPresentation = input[i];  // comment...
    i++;

    while (currentPresentation != "Finish") {
        totalPresentationsCount++;

        let presentationGrade = 0;
        for (let j = 0; j < judgesCount; j++) {
            let currentGrade = Number(input[i]);
            i++;

            presentationGrade += currentGrade;
            totalGradeSum += currentGrade;
        }

        console.log(`${currentPresentation} - ${(presentationGrade / judgesCount)}.`);

        currentPresentation = input[i];
        i++;

        // current               (i - 1)
    }


    console.log(`Student's final assessment is ${(totalGradeSum / (totalPresentationsCount * judgesCount))}.`);

}


// trainTheTrainers(["2",
//     "While-Loop",
//     "6.00",
//     "5.50",
//     "For-Loop",
//     "5.84",
//     "5.66",
//     "Finish"]);

// trainTheTrainers(["3",
//     "Finish"]);

File: Python_Courses/Python_Basics/Python_PB_2025_01/06_Nested_Loops_Exercise/04_Train_The_Trainers.py
# total_sum = 0
# total_judge_score = 0
total_avg_sum = 0
total_presentations_count = 0

judges_count = int(input())

command = input()  # 'Finish', or current_presentation, e.g. 'Arrays'
while command != 'Finish':
    current_presentation = command

    current_score = 0
    for _ in range(judges_count):
        judge_score = float(input())
        current_score += judge_score

    current_score_avg = current_score / judges_count

    print(f'{current_presentation} - {current_score_avg:.2f}.')

    # total_sum += current_score
    # total_judge_score += judges_count
    total_avg_sum += current_score_avg
    total_presentations_count += 1

    command = input()

# total_avg = total_sum / total_judge_score
total_avg = total_avg_sum / total_presentations_count
print(f'Student\'s final assessment is {total_avg:.2f}.')
